src/service/login.rs

```diff
@@ -37,3 +37,24 @@ pub fn logout(res: &mut Resources<impl RP>) -> Result<()> {
 pub fn login_exists(res: &Resources<impl RP>) -> bool {
     res.storage.get().get_token().is_some()
 }
+
+pub fn login_status(res: &Resources<impl RP>) -> Result<String> {
+    if !login_exists(res) {
+        return Ok(String::from("Login token is missing."));
+    }
+    match res.api.login_status(res.storage.get().get_token().unwrap()) {
+        Err(ApiError::PendingApiKeyError) => {
+            return Ok(String::from("Login token is being validated"))
+        }
+        Err(ApiError::ApiKeyError) => return Ok(String::from("Login token is invalid")),
```

Comment:
These texts might be somewhat unclear to the user, maybe

```suggestion
            return Ok(String::from("Login waiting to be finished in browser"))
        }
        Err(ApiError::ApiKeyError) => return Ok(String::from("Login is invalid, please log in again")),
```